src/utils/deprecated/deprecated.tsx

```diff
@@ -25,3 +45,18 @@ export const deprecated = (message: string) => {
     return DeprecatedWrapper as T;
   };
 };
+
+export const useDeprecatedPropWarning = ({
+  deprecatedProp,
+  deprecatedPropName,
+  version,
+}: IDeprecatedPropWarning): void => {
+  useEffect(() => {
+    if (deprecatedProp !== undefined) {
+      const formattedMessage = `The \`${deprecatedPropName}\` prop is deprecated and will be removed in v${version}`;
+      const deprecatedMessage = getDeprecatedMessage(formattedMessage);
+
+      console.warn(deprecatedMessage);
+    }
+  }, [deprecatedProp, deprecatedPropName, version]);
+};
```

Comment:
We could use a signature that looks something like this:

```ts
useDeprecatedProps({ props: { prop1, prop2 }, version: '2.0' });
```

That way you can use `Object.keys` on the `props` prop so you don't need to pass individual `deprecatedProp` and `deprecatedPropName`. Additionally, it would make it easier to deprecate multiple props at once.

Totally up to you, I'm not going to block this PR over this :)